lib/Sema/SemaDecl.cpp

```diff
@@ -8235,6 +8235,10 @@ void Sema::CheckMain(FunctionDecl* FD, const DeclSpec& DS) {
           Context.hasSameType(QualType(qs.strip(PT->getPointeeType()), 0),
                               Context.CharTy)) {
         qs.removeConst();
+
+        if (Context.getDefaultAS() == qs.getAddressSpace()) {
+          qs.removeAddressSpace();
+        }
```

Comment:
I can't see where it's stated explicitly, but if you look at all of the code examples you will see that there are no braces around single-statement if bodies.